backend/wmg/api/v1.py

```diff
@@ -131,10 +127,62 @@ def build_expression_summary(query_result: DataFrame) -> dict:
     return structured_result
 
 
-def extract_filter_dims_values(query_result: DataFrame) -> dict:
+def extract_filter_dims_values(query_result: DataFrame) -> Dict[str, set]:
     """
     Return unique values for each dimension in the specified query result
     """
     dims: set = set(query_result.columns) & set(cube_non_indexed_dims)
-    dim_uniq_values: dict = {dim: query_result.groupby(dim).groups.keys() for dim in dims}
+    dim_uniq_values: Dict[str, set] = {dim: query_result.groupby(dim).groups.keys() for dim in dims}
     return dim_uniq_values
+
+
```

Comment:
below moved here from query.py

Reply:
I think it might make sense to put this and the other build functions in their own file (or at least not in the api file)

Reply:
they are specifically for the API responses, so they seemed ok here.  do we have a convention that I can follow for putting api endpoint-specific logic elsewhere?

Reply:
what if make them protected with `_` prefix?

Reply:
Sorry that was more of nit, leaving them (with or without the prefix) should be okay.